api/envoy/extensions/bootstrap/internal_listener/v3/internal_listener.proto

```diff
@@ -19,4 +20,9 @@ option (xds.annotations.v3.file_status).work_in_progress = true;
 
 // Configuration for internal listener.
 message InternalListener {
+  // The internal listener client connection buffer size in KB.
+  // For example, if buffer_size is set to 5, then the actual buffer size is
+  // 5 KB = 5 * 1024 bytes.
+  // If the buffer_size is not specified, the buffer size is set to 1024 KB.
+  uint32 buffer_size = 1 [(validate.rules).uint32 = {lte: 8192}];
```

Comment:
one way is to codec the unit in the name, we can name it as `buffer_size_kb`. But we didn't use this style in other places of envoy, better to keep same naming style in envoy. so just a suggestion.

Reply:
Yeah, I thought about the same thing to use xxx_in_kb, but not finding anything similar existing thus end up not using it.  I am open to either way.

Reply:
Related, what's the reason for the upper bound number?

Reply:
Also, optional values use wrappers for ints I think.

Reply:
Sure, for changing the value to be wrappers.

Regarding the buffer upper bound number, I somehow find this doc: https://www.ibm.com/docs/en/was/9.0.5?topic=environment-tuning-tcpip-buffer-sizes, which says the default buffer is is 8kB, and maximum size is 8MB.  8KB might be a little bit small. But let me change the default size to 64kB from current 1MB.